lib/core-common/src/presets.ts

```diff
@@ -254,10 +253,13 @@ function loadPresets(
     logger.info('=> Loading presets');
   }
 
-  return presets.reduce((acc, preset) => {
-    const loaded = loadPreset(preset, level, storybookOptions);
-    return acc.concat(loaded);
-  }, []);
+  return Promise.all(
+    presets
+      .map(async (preset) => loadPreset(preset, level, storybookOptions))
+      .reduce((acc, loaded) => {
+        return acc.concat(loaded);
+      }, [])
+  );
```

Comment:
This might cause the order of operations to change!?

An preset/addon can add sub-addons and those should be evaluated before sibling addons.

Reply:
Whoops, I think I had the `Promise.all` in the wrong spot. Better now @ndelangen?